RecoEgamma/EgammaTools/src/ShowerDepth.cc

```diff
@@ -1,60 +1,29 @@
 #include "RecoEgamma/EgammaTools/interface/ShowerDepth.h"
-#include <ctgmath>
-#include <iostream>
+#include <cmath>
 
-ShowerDepth::ShowerDepth() {
-    // HGCAL average medium
-    criticalEnergy_ = 0.00536; // in GeV
-    radiationLength_ = 0.968; // in cm
 
-    // longitudinal parameters
-    // mean values
-    // shower max <T> = t0 + t1*lny
-    // <alpha> = alpha0 + alpha1*lny
-    // shower average = alpha/beta
-    meant0_ = -1.396;
-    meant1_ = 1.007;
-    meanalpha0_ = -0.0433;
-    meanalpha1_ = 0.540;
-    // sigmas
-    // sigma(lnT) = 1 /sigmalnt0 + sigmalnt1*lny;
-    // sigma(lnalpha) = 1 /sigmalnt0 + sigmalnt1*lny;
-    sigmalnt0_ = -2.506;
-    sigmalnt1_ = 1.245;
-    sigmalnalpha0_ = -0.08442;
-    sigmalnalpha1_ = 0.7904;
-    // corr(lnalpha,lnt) = corrlnalpha0_+corrlnalphalnt1_*y
-    corrlnalphalnt0_ = 0.7858;
-    corrlnalphalnt1_ = -0.0232;
-    debug_ = false;
-}
+float ShowerDepth::getClusterDepthCompatibility(float measuredDepth, float emEnergy, float& expectedDepth, float& expectedSigma) const {
+    double lny = (emEnergy > criticalEnergy_) ? std::log(emEnergy/criticalEnergy_) : 0.;
 
-float ShowerDepth::getClusterDepthCompatibility(float length, float emEnergy,float & expectedDepth, float & expectedSigma) const {
-    float lny = emEnergy/criticalEnergy_>1. ? std::log(emEnergy/criticalEnergy_) : 0.;
     // inject here parametrization results
-    float meantmax = meant0_ + meant1_*lny;
-    float meanalpha = meanalpha0_ + meanalpha1_*lny;
-    float sigmalntmax = 1.0 / (sigmalnt0_+sigmalnt1_*lny);
-    float sigmalnalpha = 1.0 / (sigmalnalpha0_+sigmalnalpha1_*lny);
-    float corrlnalphalntmax = corrlnalphalnt0_+corrlnalphalnt1_*lny;
-
-    float invbeta = meantmax/(meanalpha-1.);
-    float predictedLength = meanalpha*invbeta;
-    predictedLength *= radiationLength_;
+    double meantmax = meant0_ + meant1_*lny;
```

Comment:
was float -> double required by some precision considerations?
float can be faster